yeonjae/programmers-2/모음사전.js

```diff
@@ -0,0 +1,29 @@
+function solution(word) {
+  const dic = ["A", "E", "I", "O", "U"];
+  let checkValid = true;
+  let answer = -1;
+
+  const dfs = (str) => {
+    if (checkValid) {
+      answer++;
+
+      if (str === word) {
+        checkValid = false;
+      } else if (str.length < 5) {
+        dic.forEach((el) => dfs(str + el));
+      }
+      return;
```

Comment:
```suggestion
  if (str === word) {
    isValid = false;
    return;
  }

  if (str.length < 5) {
    dic.forEach((el) => dfs(str + el));
  }
```
- return문을 상단에 둠으로써 else if문을 제거하는게 어떨까요

Reply:
그렇네요! 분리함으로써 이게 좀 가독성이 좋은것같습니다!